### DISCUSSION TITLE
Automatically add appropriate decorators when auto-completing method in child class

### DISCUSSION THREAD

**Discussion by Holt59** [microsoft/pylance-release#4343] [Enhancement]
When inheriting a class, Pylance can generates a "default body" when overriding parent class. If I auto-complete the following, it will generate a matching function in `B`.

```python
class A:
    def my_method(self, a: int, b: float) -> str:
        ...

class B(A):
    def my_
```

This auto-completion does not automatically adds appropriate decorators, in particular, `@property` to the generated function. It would be nice if this was done automatically.

**Comment by rchiodo** [comment]
Thanks for the suggestion. 

I want to make sure I understand. 

If I have say class A like so:

```python
class A:
   @property
   def prop1() -> str:
       return 4
```

And I override the property in the sub class, you would expect `@property` to be added?

```python
class B(A):
    def prop1
```

**Comment by Holt59** [comment]
@rchiodo Yes, currently if I auto-complete (Ctrl+Space for me) after typing `def prop1`, I will get:

```python
def prop1() -> str:
    pass
```

I'd like to get

```python
@property  # automatically added
def prop1() -> str:
    pass
```

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
